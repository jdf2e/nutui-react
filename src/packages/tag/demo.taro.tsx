import React from 'react'
import Taro from '@tarojs/taro'
import { CircleClose } from '@nutui/icons-react-taro'
import { Cell, Tag } from '@/packages/nutui.react.taro'
import Header from '@/sites/components/header'
import { useTranslate } from '@/sites/assets/locale/taro'

const TagDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      tag: '标签',
      basic: '基础用法',
      style: '样式风格',
      customColor: '自定义颜色',
      plain: '空心样式',
      round: '圆角样式',
      mark: '标记样式',
      closeable: '可关闭标签',
      backgroundColor: '背景颜色',
      textColor: '文字颜色',
      plainColor: '空心颜色',
    },
    'en-US': {
      tag: 'Tag',
      basic: 'Basic Usage',
      style: 'Style',
      customColor: 'Custom Color',
      plain: 'Plain',
      round: 'Round',
      mark: 'Mark',
      closeable: 'Closeable',
      backgroundColor: 'Background Color',
      textColor: 'Text Color',
      plainColor: 'Plain Color',
    },
  })
  return (
    <>
      <Header />
      <div className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''}`}>
        <h2>{translated.basic}</h2>
        <Cell.Group>
          <Cell
            title="primary"
            extra={<Tag type="primary">{translated.tag}</Tag>}
          />
          <Cell
            title="success"
            extra={<Tag type="success">{translated.tag}</Tag>}
          />
          <Cell
            title="danger"
            extra={<Tag type="danger">{translated.tag}</Tag>}
          />
          <Cell
            title="warning"
            extra={<Tag type="warning">{translated.tag}</Tag>}
          />
        </Cell.Group>

        <h2>{translated.style}</h2>
        <Cell.Group>
          <Cell
            title={translated.plain}
            extra={<Tag plain>{translated.tag}</Tag>}
          />
          <Cell
            title={translated.round}
            extra={
              <Tag round type="primary">
                {translated.tag}
              </Tag>
            }
          />
          <Cell
            title={translated.mark}
            extra={
              <Tag mark type="primary">
                {translated.tag}
              </Tag>
            }
          />
          <Cell
            title={translated.closeable}
            extra={
              <Tag
                closeable
                closeIcon={<CircleClose width={12} height={12} />}
                onClose={() => Taro.showToast({ title: 'Tag closed' })}
                type="primary"
              >
                {translated.tag}
              </Tag>
            }
          />
        </Cell.Group>

        <h2>{translated.customColor}</h2>
        <Cell.Group>
          <Cell
            title={translated.backgroundColor}
            extra={<Tag background="#FA685D">{translated.tag}</Tag>}
          />
          <Cell
            title={translated.textColor}
            extra={
              <Tag background="#E9E9E9" color="#999999">
                {translated.tag}
              </Tag>
            }
          />
          <Cell
            title="空心颜色"
            extra={
              <Tag background="#FA2400" plain>
                {translated.tag}
              </Tag>
            }
          />
        </Cell.Group>
      </div>
    </>
  )
}

export default TagDemo
