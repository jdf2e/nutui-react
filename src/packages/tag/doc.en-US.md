# Tag

### introduce

Label for labeling and classification.

### Install

``` javascript
// react
import { Tag } from '@nutui/nutui-react';

```

## Code instance

### Basic usage

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag type="primary">Label</Tag>
      <Tag type="success">Label</Tag>
      <Tag type="danger">Label</Tag>
      <Tag type="warning">Label</Tag>
    </>
  )
}
export default App;
```

:::

### Hollow style

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag plain>Label</Tag>
    </>
  )
}
export default App;
```

:::

### Rounded style

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag round type="primary">Label</Tag>
    </>
  )
}
export default App;
```

:::

### Label style

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag mark type="primary">Label</Tag>
    </>
  )
}
export default App;
```

:::

### Can close label

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag  closeable onClose={()=>alert('Tag closed')}  type="primary">标签</Tag>
    </>
  )
}
export default App;
```

:::

### Custom color

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag color="#FA685D">Label</Tag>
      <Tag color="#E9E9E9" textColor="#999999">Label</Tag>
      <Tag color="#FA2400" plain>Label</Tag>
    </>
  )
}
export default App;
```

:::

### Click event

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag type='primary' onClick={()=>alert('Tag clicked')}>Tag</Tag>
    </>
  )
}
export default App;
```

:::

### Display control

:::demo

```tsx
import React, {useState} from "react";
import { Tag,Button } from '@nutui/nutui-react';

const App = () => {
  const  [isShow,setIsShow] = useState(true) // Whether to display a tag component
  return (
    <>
    {
      isShow? (
        <Tag type='primary' onClick={()=>alert('Tag clicked')}>Label</Tag>
      ):null
    }  
    <Button type='default' size="small" onClick={()=>{setIsShow(false)}} >delete</Button>
    </>
  )
  
}
export default App;
```

:::

## API

### Props

| Field       | illustrate                                      | type    | Defaults    |
|------------|--------------------------------------------------|---------|-----------|
| type       | Label type, the optional value is primary success danger warning | String  | `default` |
| color      | Label color                                         | String  | -         |
| texColor | Text color, priority is higher than the color attribute  | String  | `white`   |
| plain      | Whether it is hollow                               | Boolean | `false`   |
| round      | Whether it is a rounded style                      | Boolean | `false`   |
| mark       | Whether it is a tag style                          | Boolean | `false`   |
| closeable  | Whether it can be closed label                     | Boolean | `false`   |
| iconSize`v1.4.7` | the size of closed label  | string | number | ` 12px`|

### Event

| Event name | illustrate  | Callback parameter |
|----------|----------|----------|
| onClick    | Click event | event    |
| onClose    | Close event | event    |


## Theming

### CSS Variables

The component provides the following CSS variables, which can be used to customize styles. Please refer to [ConfigProvider component](#/en-US/component/configprovider).

| Name | Default Value |
| --- | --- |
| --nutui-tag-padding | ` 0 4px`|
| --nutui-tag-font-size | ` 12px` |
| --nutui-tag-default-border-radius | `  4px` |
| --nutui-tag-round-border-radius | ` 8px` |
| --nutui-tag-default-background-color | `  #000000` |
| --nutui-tag-primary-background-color | `  #3460fa` |
| --nutui-tag-success-background-color | `  #4fc08d` |
| --nutui-tag-danger-background-color-plain | `  #df3526` |
| --nutui-tag-warning-background-color | `  #f3812e` |
| --nutui-tag-default-color | ` #ffffff` |
| --nutui-tag-border-width | ` 1px` |
| --nutui-tag-height | ` auto` |
