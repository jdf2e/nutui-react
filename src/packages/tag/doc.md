# Tag 标签

## 介绍

用于标记和分类的标签。

## 安装

```tsx
import { Tag } from '@nutui/nutui-react';

```

## 代码演示

### 基础用法

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag type="primary">标签</Tag>
      <Tag type="success">标签</Tag>
      <Tag type="info">标签</Tag>
      <Tag type="danger">标签</Tag>
      <Tag type="warning">标签</Tag>
    </>
  )
}
export default App;
```

:::

### 空心样式

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag plain>标签</Tag>
    </>
  )
}
export default App;
```

:::

### 圆角样式

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag round type="primary">标签</Tag>
    </>
  )
}
export default App;
```

:::

### 标记样式

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag mark type="primary">标签</Tag>
    </>
  )
}
export default App;
```

:::

### 可关闭标签

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag  closeable onClose={()=>alert('Tag closed')}  type="primary">标签</Tag>
    </>
  )
}
export default App;
```

:::

### 自定义颜色

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag background="#FA685D">标签</Tag>
      <Tag background="#E9E9E9" color="#999999">标签</Tag>
      <Tag background="#FA2400" plain>标签</Tag>
    </>
  )
}
export default App;
```

:::

### 点击事件

:::demo

```tsx
import React from "react";
import { Tag } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Tag type='primary' onClick={()=>alert('Tag clicked')}>标签</Tag>
    </>
  )
}
export default App;
```

:::

### 展示控制

:::demo

```tsx
import React, {useState} from "react";
import { Tag,Button } from '@nutui/nutui-react';

const App = () => {
  const  [isShow,setIsShow] = useState(true) // 是否展示Tag组件
  return (
    <>
    {
      isShow? (
        <Tag type='primary' onClick={()=>alert('Tag clicked')}>标签</Tag>
      ):null
    }  
    <Button type='default' size="small" onClick={()=>{setIsShow(false)}} >点击删除Tag</Button>
    </>
  )
  
}
export default App;
```

:::

## Tag

### Props

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| type | 标签类型 | `primary` \| `success` \| `danger` \| `warning` | `default` |
| background | 标签颜色 | `string` | `-` |
| color | 文本颜色，优先级高于color属性 | `string` | `white` |
| plain | 是否为空心样式 | `boolean` | `false` |
| round | 是否为圆角样式 | `boolean` | `false` |
| mark | 是否为标记样式 | `boolean` | `false` |
| closeable | 是否为可关闭标签 | `boolean` | `false` |
| closeIcon | 关闭按钮 | `ReactNode` | `null` |
| onClick | 点击事件 | `(e: MouseEvent<HTMLDivElement>) => void` | `-` |
| onClose | 关闭事件 | `(e?: any) => void` | `-` |

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称 | 说明 | 默认值 |
| --- | --- | --- |
| \--nutui-tag-padding | padding 值 | `0 4px` |
| \--nutui-tag-font-size | 字号 | `12px` |
| \--nutui-tag-border-radius | 圆角 | `4px` |
| \--nutui-tag-height | 高度 | `auto` |
| \--nutui-tag-color | 字色 | `#ffffff` |
| \--nutui-tag-border-width | 边宽 | `1px` |
| \--nutui-tag-background-color | 背景色 | `#000000` |
| \--nutui-tag-primary-background-color | 主色背景色 | `#3460fa` |
| \--nutui-tag-success-background-color | 成功背景色 | `#4fc08d` |
| \--nutui-tag-info-background-color | 信息背景色 | `$brand-info-color` |
| \--nutui-tag-warning-background-color | 警告背景色 | `#f3812e` |
| \--nutui-tag-round-border-radius | round模式下的圆角 | `8px` |
| \--nutui-tag-mark-border-radius | mark模式下的圆角 | `0 12px 12px 0` |