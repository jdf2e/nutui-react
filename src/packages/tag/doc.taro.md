# Tag 标签

## 介绍

用于标记和分类的标签。

## 安装

```tsx
import { Tag } from '@nutui/nutui-react-taro';
```

## 代码实例

### 基础用法

:::demo

<CodeBlock src='taro/demo1.tsx'></CodeBlock>

:::

### 样式风格

:::demo

<CodeBlock src='taro/demo2.tsx'></CodeBlock>

:::

### 自定义颜色

:::demo

<CodeBlock src='taro/demo3.tsx'></CodeBlock>

:::

### 自定义颜色

:::demo

<CodeBlock src='taro/demo4.tsx'></CodeBlock>

:::

## Tag

### Props

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| type | 标签类型 | `primary` \| `default` \| `info` \| `success` \| `danger` \| `warning` | `default` |
| background | 标签颜色 | `string` | `-` |
| color | 文本颜色，优先级高于color属性 | `string` | `white` |
| plain | 是否为空心样式 | `boolean` | `false` |
| round | 是否为圆角样式 | `boolean` | `false` |
| mark | 是否为标记样式 | `boolean` | `false` |
| closeable | 是否为可关闭标签 | `boolean` | `false` |
| closeIcon | 关闭按钮 | `ReactNode` | `null` |
| onClick | 点击事件 | `(e: MouseEvent) => void` | `-` |
| onClose | 关闭事件 | `(e?: any) => void` | `-` |

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称 | 说明 | 默认值 |
| --- | --- | --- |
| \--nutui-tag-padding | padding 值 | `0 2px` |
| \--nutui-tag-font-size | 字号 | `10px` |
| \--nutui-tag-border-radius | 圆角 | `2px` |
| \--nutui-tag-height | 高度 | `14px` |
| \--nutui-tag-color | 字色 | `#ffffff` |
| \--nutui-tag-border-width | 边宽 | `1px` |
| \--nutui-tag-background-color | 背景色 | `$color-title` |
| \--nutui-tag-primary-background-color | 主色背景色 | `$color-primary-gradient-1` |
| \--nutui-tag-success-background-color | 成功背景色 | `#4fc08d` |
| \--nutui-tag-info-background-color | 信息背景色 | `$color-info` |
| \--nutui-tag-warning-background-color | 警告背景色 | `#f3812e` |
| \--nutui-tag-danger-background-color | 危险背景色 | `$color-primary` |
| \--nutui-tag-round-border-radius | round模式下的圆角 | `8px` |
| \--nutui-tag-mark-border-radius | mark模式下的圆角 | `0 8px 8px 0` |