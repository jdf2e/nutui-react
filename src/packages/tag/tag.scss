@import '../icon/icon.scss';
.nut-tag {
  display: inline;
  padding: 0 4px;
  border-radius: 4px;
  ._icon {
    vertical-align: middle;
    margin-left: 4px;
  }

  @mixin plain {
    &.nut-tag--plain {
      color: $tag-default-background-color;
      border: $tag-border-width solid $tag-default-background-color;
    }
  }

  &--default {
    color: $tag-default-color;
    background: $tag-default-background-color;
    border: $tag-border-width solid transparent;
    @include plain;
  }

  &--primary {
    color: $tag-default-color;
    background: $tag-primary-background-color;
    border: $tag-border-width solid transparent;
    @include plain;
  }

  &--success {
    color: $tag-default-color;
    background: $tag-success-background-color;
    border: $tag-border-width solid transparent;
    @include plain;
  }

  &--danger {
    color: $tag-default-color;
    background: $tag-danger-background-color;
    border: $tag-border-width solid transparent;
    @include plain;
  }

  &--warning {
    color: $tag-default-color;
    background: $tag-warning-background-color;
    border: $tag-border-width solid transparent;
    @include plain;
  }
  &--round {
    border-radius: 8px;
  }
  &--mark {
    border-radius: 0 12px 12px 0;
  }
  &--close {
    margin-left: 4px;
    cursor: pointer;
  }
}
