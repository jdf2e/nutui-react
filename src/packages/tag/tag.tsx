import React, {
  CSSProperties,
  FunctionComponent,
  useEffect,
  useState,
} from 'react'
import Icon from '@/packages/icon'

import { IComponent, ComponentDefaults } from '@/utils/typings'

export interface TagProps extends IComponent {
  type: TagType
  color: string
  textColor: string
  plain: boolean
  round: boolean
  mark: boolean
  closeable: boolean
  prefixCls: string
  onClick: (e: MouseEvent) => void
  onClose: (e?: any) => void
  children?: React.ReactNode
}

export type TagType = 'default' | 'primary' | 'success' | 'warning' | 'danger'

const defaultProps = {
  ...ComponentDefaults,
  type: 'default',
  color: '',
  textColor: '',
  plain: false,
  round: false,
  mark: false,
  closeable: false,
  prefixCls: 'nut-tag',
  onClose: (e: any) => {},
  onClick: (e: MouseEvent) => {},
} as TagProps
export const Tag: FunctionComponent<Partial<TagProps>> = (props) => {
  const {
    className,
    style,
    color,
    plain,
    type,
    round,
    prefixCls,
    children,
    mark,
    closeable,
    textColor,
    onClick,
    onClose,
    iconClassPrefix,
    iconFontClassName,
  } = {
    ...defaultProps,
    ...props,
  }
  const [btnName, setBtnName] = useState('')
  const [isTagShow, setIsTagShow] = useState(true)
  useEffect(() => {
    setBtnName(classes())
  }, [
    type,
    color,
    textColor,
    plain,
    round,
    mark,
    closeable,
    prefixCls,
    onClick,
    onClose,
  ])
  const classes = () => {
    const prefixCls = 'nut-tag'
    return `${prefixCls}
    ${type ? `${prefixCls}--${type}` : ''}
    ${plain ? `${prefixCls}--plain` : ''}
    ${round ? `${prefixCls}--round` : ''}
    ${mark ? `${prefixCls}--mark` : ''}
    ${closeable ? `${prefixCls}--close` : ''}`
  }
  const handleClick = (e: any) => {
    if (props.onClick) {
      props.onClick(e)
    }
  }
  const getStyle = () => {
    const style: CSSProperties = {}
    if (textColor) {
      style.color = textColor
      if (plain) {
        style.background = '#fff'
      } else if (color) {
        style.background = color
      }
    } else if (color) {
      style.color = '#fff'
      style.background = color
    }
    return style
  }
  return (
    <>
      {closeable ? (
        isTagShow && (
          <div
            className={`${btnName} ${className}`}
            style={{ ...style, ...getStyle() }}
            onClick={(e) => handleClick(e)}
          >
            {children && <span className="text">{children}</span>}
            <Icon
              classPrefix={iconClassPrefix}
              fontClassName={iconFontClassName}
              className="_icon"
              name="close"
              size="12"
              onClick={(e) => {
                setIsTagShow(false)
                if (props.onClose) {
                  props.onClose(e)
                }
              }}
            />
          </div>
        )
      ) : (
        <div
          className={`${btnName} ${className}`}
          style={{ ...style, ...getStyle() }}
          onClick={(e) => handleClick(e)}
        >
          {children && <span className="text">{children}</span>}
        </div>
      )}
    </>
  )
}

Tag.defaultProps = defaultProps
Tag.displayName = 'NutTag'
