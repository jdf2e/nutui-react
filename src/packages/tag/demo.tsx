import React from 'react'
import { Tag } from './tag'
import Cell from '@/packages/cell'
import CellGroup from '@/packages/cellgroup'

const TagDemo = () => {
  return (
    <>
      <div className="demo">
        <h2>基础用法</h2>
        <CellGroup>
          <Cell title="primary类型" linkSlot={<Tag type="primary">标签</Tag>} />
          <Cell title="success类型" linkSlot={<Tag type="success">标签</Tag>} />
          <Cell title="danger类型" linkSlot={<Tag type="danger">标签</Tag>} />
          <Cell title="warning类型" linkSlot={<Tag type="warning">标签</Tag>} />
        </CellGroup>

        <h2>样式风格</h2>
        <CellGroup>
          <Cell title="空心样式" linkSlot={<Tag plain>标签</Tag>} />
          <Cell
            title="圆角样式"
            linkSlot={
              <Tag round type="primary">
                标签
              </Tag>
            }
          />
          <Cell
            title="标记样式"
            linkSlot={
              <Tag mark type="primary">
                标签
              </Tag>
            }
          />
          <Cell
            title="可关闭标签"
            linkSlot={
              <Tag
                closeable
                iconSize="14px"
                onClose={() => alert('Tag closed')}
                type="primary"
              >
                标签
              </Tag>
            }
          />
        </CellGroup>

        <h2>自定义</h2>
        <CellGroup>
          <Cell title="背景颜色" linkSlot={<Tag color="#FA685D">标签</Tag>} />
          <Cell
            title="文字颜色"
            linkSlot={
              <Tag color="#E9E9E9" textColor="#999999">
                标签
              </Tag>
            }
          />
          <Cell
            title="空心颜色"
            linkSlot={
              <Tag color="#FA2400" plain>
                标签
              </Tag>
            }
          />
        </CellGroup>
      </div>
    </>
  )
}

export default TagDemo
