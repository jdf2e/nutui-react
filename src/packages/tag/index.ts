import { Tag } from './tag'

export type { TagType, TagProps } from './tag'
export default Tag
