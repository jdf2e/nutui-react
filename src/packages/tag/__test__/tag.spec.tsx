import * as React from 'react'

import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Tag, TagType } from '../tag'

test('color test', () => {
  const state = {
    color: 'rgb(250, 104, 93)',
  }
  const { getByTestId, container } = render(<Tag color={state.color}>TEST</Tag>)
  expect(container.querySelector('.nut-tag--default')).toHaveAttribute(
    'style',
    'color: rgb(255, 255, 255); background: rgb(250, 104, 93);'
  )
})
test('type test', () => {
  const state: {
    type: TagType
  } = {
    type: 'primary',
  }
  const { container } = render(<Tag type={state.type}>TEST</Tag>)
  if (container.firstChild) {
    expect(container).toMatchSnapshot()
  }
})

test('plain test', () => {
  const state = {
    plain: true,
  }
  const { container } = render(<Tag plain={state.plain}>TEST</Tag>)
  const el = container.querySelectorAll('.nut-tag--plain').length
  expect(el > 0).toBe(true)
})

test('round test', () => {
  const state = {
    round: true,
  }
  const { container } = render(<Tag round={state.round}>TEST</Tag>)
  const el: any = container.querySelectorAll('.nut-tag--round').length
  expect(el > 0).toBe(true)
})

test('mark test', () => {
  const state = {
    mark: true,
  }
  const { container } = render(<Tag mark={state.mark}>TEST</Tag>)
  const el: any = container.querySelectorAll('.nut-tag--mark')
  expect(el.length > 0).toBe(true)
})

test('closeable && onClose  test', () => {
  const state = {
    closeable: true,
  }
  const handleClose = jest.fn()
  const { container } = render(
    <>
      <h1 className="text">0</h1>
      <Tag closeable={state.closeable} onClose={handleClose}>
        TEST
      </Tag>
    </>
  )
  const el: any = container.querySelectorAll('.nut-tag--close')[0]
  const icon: any = el.childNodes[1]
  fireEvent.click(icon)
  expect(el && el.length > 0).toBe(false)
  expect(handleClose).toBeCalled()
})

test('emit click event', () => {
  const testClick = jest.fn()
  function tree() {
    return render(
      <Tag data-testid="tag-click" onClick={() => testClick()} prefixCls="test">
        TEST
      </Tag>
    )
  }
  const { container } = tree()
  const el: any = container.querySelector('test')
  if (el) {
    fireEvent.click(el)
    expect(testClick).toBeCalled()
  }
})
