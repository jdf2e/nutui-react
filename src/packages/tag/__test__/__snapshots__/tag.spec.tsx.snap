// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type test 1`] = `
<div>
  <div>
    <div
      class="nut-tag
    nut-tag--primary
    
    
    
    "
    >
      <span
        class="text"
      >
        TEST
      </span>
    </div>
  </div>
</div>
`;

